lib/dockerregistry/inventory.go

```diff
@@ -1171,6 +1236,143 @@ func (sc *SyncContext) ReadRegistries(
 	sc.ExecRequests(populateRequests, processRequest)
 }
 
+// ReadGManifestLists reads all manifest lists and populates the ParentDigest
```

Comment:
_If_ we could afford to give up parallel execution, this function would be much easier to understand!

Reply:
Yes... I can think of a way to simplify it but let's hold off on that until later (will add a TODO for myself here). Ack.